File: events/stickyManager.js
const fs = require('fs');
const path = require('path');

const STORAGE_PATH = path.join(__dirname, '../settings/stickyMessages.json');
const locks = new Set();

function readData() {
    try {
        if (!fs.existsSync(STORAGE_PATH)) {
            return {};
        }
        const data = fs.readFileSync(STORAGE_PATH, 'utf8');
        return JSON.parse(data);
    } catch (error) {
        console.error('Error reading sticky messages data:', error);
        return {};
    }
}

function writeData(data) {
    try {
        fs.writeFileSync(STORAGE_PATH, JSON.stringify(data, null, 2));
    } catch (error) {
        console.error('Error writing sticky messages data:', error);
    }
}

module.exports = {
    getSticky: (guildId, channelId) => {
        const data = readData();
        return data[guildId]?.[channelId];
    },
    getAllStickies: (guildId) => {
        const data = readData();
        return data[guildId] || {};
    },
    setSticky: (guildId, channelId, config) => {
        const data = readData();
        if (!data[guildId]) data[guildId] = {};
        data[guildId][channelId] = config;
        writeData(data);
    },
    removeStickyByChannel: (guildId, channelId) => {
        const data = readData();
        if (data[guildId]) {
            delete data[guildId][channelId];
            writeData(data);
        }
    },
    removeStickyByName: (guildId, name) => {
        const data = readData();
        if (data[guildId]) {
            let found = false;
            for (const channelId in data[guildId]) {
                if (data[guildId][channelId].name === name) {
                    delete data[guildId][channelId];
                    found = true;
                    writeData(data);
                    return true;
                }
            }
        }
        return false;
    },
    updateLastMessageId: (guildId, channelId, messageId) => {
        const data = readData();
        if (data[guildId] && data[guildId][channelId]) {
            data[guildId][channelId].lastMessageId = messageId;
            writeData(data);
        }
    },
    isLocked: (channelId) => locks.has(channelId),
    lock: (channelId) => locks.add(channelId),
    unlock: (channelId) => locks.delete(channelId)
};

File: utils/sticky_manager.py
import json
import os

STORAGE_PATH = 'stickyMessages.json'
locks = set()

class StickyManager:
    @staticmethod
    def read_data():
        if not os.path.exists(STORAGE_PATH):
            return {}
        try:
            with open(STORAGE_PATH, 'r') as f:
                return json.load(f)
        except Exception as e:
            print(f"Error reading sticky data: {e}")
            return {}

    @staticmethod
    def write_data(data):
        try:
            with open(STORAGE_PATH, 'w') as f:
                json.dump(data, f, indent=4)
        except Exception as e:
            print(f"Error writing sticky data: {e}")

    @classmethod
    def get_sticky(cls, guild_id, channel_id):
        data = cls.read_data()
        return data.get(str(guild_id), {}).get(str(channel_id))

    @classmethod
    def set_sticky(cls, guild_id, channel_id, config):
        data = cls.read_data()
        g_id = str(guild_id)
        if g_id not in data:
            data[g_id] = {}
        data[g_id][str(channel_id)] = config
        cls.write_data(data)

    @classmethod
    def remove_sticky(cls, guild_id, channel_id):
        data = cls.read_data()
        g_id = str(guild_id)
        if g_id in data and str(channel_id) in data[g_id]:
            del data[g_id][str(channel_id)]
            cls.write_data(data)

    @classmethod
    def update_last_message_id(cls, guild_id, channel_id, message_id):
        data = cls.read_data()
        g_id = str(guild_id)
        c_id = str(channel_id)
        if g_id in data and c_id in data[g_id]:
            data[g_id][c_id]['lastMessageId'] = message_id
            cls.write_data(data)

    @staticmethod
    def is_locked(channel_id):
        return channel_id in locks

    @staticmethod
    def lock(channel_id):
        locks.add(channel_id)

    @staticmethod
    def unlock(channel_id):
        if channel_id in locks:
            locks.remove(channel_id)
